src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Runtime/ConstructorParametersShouldMatchPropertyNames.cs

```diff
@@ -0,0 +1,194 @@
+                                                                             s_localizableMessageField,
+                                                                             DiagnosticCategory.Design,
+                                                                             RuleLevel.BuildWarning,
+                                                                             description: s_localizableDescription,
+                                                                             isPortedFxCopRule: false,
+                                                                             isDataflowRule: false);
+
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(PropertyRule, FieldRule);
+
+        public override void Initialize(AnalysisContext context)
+        {
+            context.EnableConcurrentExecution();
+            context.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.None);
+
+            context.RegisterCompilationStartAction(
+                (compilationStartContext) =>
+                {
+                    INamedTypeSymbol? jsonConstructorAttributeNamedSymbol = compilationStartContext.Compilation.GetOrCreateTypeByMetadataName(WellKnownTypeNames.SystemTextJsonSerializationJsonConstructorAttribute);
+                    if (jsonConstructorAttributeNamedSymbol == null)
+                    {
+                        return;
+                    }
+
+                    var paramAnalyzer = new ParameterAnalyzer(jsonConstructorAttributeNamedSymbol);
+
+                    compilationStartContext.RegisterOperationBlockStartAction(
+                        (startOperationBlockContext) =>
+                        {
+                            if (startOperationBlockContext.OwningSymbol is IMethodSymbol method
+                                && !paramAnalyzer.ShouldAnalyzeMethod(method))
+                            {
+                                return;
+                            }
+
+                            startOperationBlockContext.RegisterOperationAction(
+                                context => ParameterAnalyzer.AnalyzeOperationAndReport(context),
+                                OperationKind.ParameterReference);
+                        });
+                });
+        }
+
+        private sealed class ParameterAnalyzer
+        {
+            private readonly INamedTypeSymbol _jsonConstructorAttributeInfoType;
+
+            public ParameterAnalyzer(INamedTypeSymbol jsonConstructorAttributeInfoType)
+            {
+                _jsonConstructorAttributeInfoType = jsonConstructorAttributeInfoType;
+            }
+
+            public static void AnalyzeOperationAndReport(OperationAnalysisContext context)
+            {
+                var operation = (IParameterReferenceOperation)context.Operation;
+
+                if (operation.Parent is not IAssignmentOperation assignment)
```

Comment:
This is likely to fail for tuple assignments like `(_x, _y) = (x, y);`

Reply:
I'm not sure whether I've completely got your point, but it seems that tuple assignment works fine with it, see the following commit: [[POC] Verify tuple assignment is working with CA1071](https://github.com/dotnet/roslyn-analyzers/pull/4877/commits/5a625475e610b85c312c04588feda0d4104ccc51)

Field names you've mentioned do not match parameter names, `JsonConstructor` will throw a runtime exception. That's why diagnostic must still be reported anyway for this case.

Reply:
@psxvoid The tests in that commit are no diagnostic tests. I meant to confirm that diagnostic cases are working correctly for tuple assignments.

Reply:
Yep, sorry, it doesn't work with tuple assignments. I'll come back to it a bit later.

Reply:
Done.